File: .sling/libs/sushi/eqela/jk-data/0.27.0/jk.generator.lua
jk = jk or {}

jk.generator = jk.generator or {}

jk.generator.DataGenerator = {}
jk.generator.DataGenerator.__index = jk.generator.DataGenerator
_vm:set_metatable(jk.generator.DataGenerator, {})

function jk.generator.DataGenerator._create()
	local v = _vm:set_metatable({}, jk.generator.DataGenerator)
	return v
end

function jk.generator.DataGenerator:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.generator.DataGenerator'
	self['_isType.jk.generator.DataGenerator'] = true
end

function jk.generator.DataGenerator:_construct0()
	jk.generator.DataGenerator._init(self)
	return self
end

function jk.generator.DataGenerator:generateCodeOrId(length, allDigits, allCaps, prefix)
	local random = _g.jk.random.Random:getSharedInstance()
	local sb = _g.jk.lang.StringBuilder._construct0(_g.jk.lang.StringBuilder._create())
	if _g.jk.lang.String:isNotEmpty(prefix) then
		do sb:appendString(prefix) end
	end
	do
		local l = length
		if l < 1 then
			l = 1
		end
		do
			local i = 0
			while i < l do
				if allDigits then
					do sb:appendCharacter(random:nextInt3(48, 58)) end
				elseif random:nextInt3(0, 2) == 0 then
					do sb:appendCharacter(random:nextInt3(48, 58)) end
				elseif allCaps then
					do sb:appendCharacter(_g.jk.lang.Character:toUppercase(random:nextInt3(97, 123))) end
				elseif random:nextInt3(0, 2) == 0 then
					do sb:appendCharacter(_g.jk.lang.Character:toUppercase(random:nextInt3(97, 123))) end
				else
					do sb:appendCharacter(random:nextInt3(97, 123)) end
				end
				do i = i + 1 end
			end
		end
		do return sb:toString() end
	end
end

File: .sling/libs/web5/eqela/jk-data/0.32.0/jk.generator.js
let JkGeneratorDataGenerator = function() {};

JkGeneratorDataGenerator.prototype._t = { "JkGeneratorDataGenerator" : true };
JkGeneratorDataGenerator.prototype._tobj = JkGeneratorDataGenerator;

JkGeneratorDataGenerator.NEW = function() {
	var v = new JkGeneratorDataGenerator;
	return v.CTOR_JkGeneratorDataGenerator();
};

JkGeneratorDataGenerator.prototype.CTOR_JkGeneratorDataGenerator = function() {
	return this;
};

JkGeneratorDataGenerator.generateCodeOrId = function(_length, allDigits, allCaps, prefix) {
	var random = JkRandomRandom.getSharedInstance();
	var sb = JkLangStringBuilder.NEW();
	if(JkLangString.isNotEmpty(prefix)) {
		sb.appendString(prefix);
	}
	var l = _length;
	if(l < 1) {
		l = 1;
	}
	for(var i = 0 ; i < l ; i++) {
		if(allDigits) {
			sb.appendCharacter((random.nextInt3(48, 58)));
		}
		else if(random.nextInt3(0, 2) == 0) {
			sb.appendCharacter((random.nextInt3(48, 58)));
		}
		else if(allCaps) {
			sb.appendCharacter((JkLangCharacter.toUppercase((random.nextInt3(97, 123)))));
		}
		else if(random.nextInt3(0, 2) == 0) {
			sb.appendCharacter((JkLangCharacter.toUppercase((random.nextInt3(97, 123)))));
		}
		else {
			sb.appendCharacter((random.nextInt3(97, 123)));
		}
	}
	return sb.toString();
};

JkGeneratorDataGenerator.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkGeneratorDataGenerator"] === true;
};
